xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -167,10 +181,20 @@ private static ManagedChannel initLbChannel(Helper helper, String balancerName)
               uoe);
           logger.log(
               FINEST,
-              "creating oob channel for target {0} using default ManagedChannelBuilder",
+              "creating oob channel for target {0} normally",
               balancerName);
         }
-        channel = ManagedChannelBuilder.forTarget(balancerName).build();
+
+        // Use the first supported channel credentials configuration.
+        // Currently, only "google_default" is supported.
+        for (ChannelCreds creds : channelCredsList) {
+          if (creds.getType().equals("google_default")) {
+            channel = GoogleDefaultChannelBuilder.forTarget(balancerName).build();
```

Comment:
Don't explicitly depend on alts. Instead, define `AltsChannelProvider` in `io.grpc.alts` similar to `NettyChannelProvider` but has priority prior to netty. Users will add needed dependencies based on their environment. For example, users can have their own custom channel builder like `CloudToProdChannelBuilder` and custom channel provider, and they can add their custom dependencies.

Reply:
But we do not want to build channel with `GoogleDefaultChannelBuilder` by default, even if `grpc-alts` is in the runtime dependency with a `AltsChannelProvider`. We want to do this based on bootstrap file's configuration.

Reply:
@ejona86 Do we want to only and explicitly use `GoogleDefaultChannelBuilder` when "google_default" is specified?

Reply:
As far as I understand the design and how this is implemented in other languages, the only currently supported option here is `google_default` which should explicitly use `GoogleDefaultChannelBuilder` (the other option is to not specify anything, which would result in inheriting the config of the parent channel). I don't immediately see a use case for allowing users to register their own customer channel builder that the xDS LB channel would pick up via the provider mechanism when the bootstrap file specifies `google_default` - I think this would qualify as a "reserved" keyword?

Reply:
Regarding making `google_default` keyword, I was thinking about creating an Enum for `ChannelCreds.type` (which we currently use `String`) so that each `ChannelCreds` has its `type` converted to Enum value when it is parsed. Does this sound good?

Reply:
Enum type may not have backward compatibility.

Reply:
Uh.. if it's not a public API and only use for client side, then there is no backward compatibility issue.

Reply:
Where should this enum go? I am considering `GrpcUtil.java`

Reply:
I thought it was "compute_engine_creds" but if it is "google_default" that is fine. But yes, we only need GoogleDefaultChannelBuilder (or similar) in that one case. If there is no credential specified, we need to use the `Helper.createResolvingOobChannel`. And if it is a value we don't understand, we should error.

We need to fallback to createResolvingOobChannel for the LOAS2 case internally.

The long-term design is to have a registry of call/channel credentials that we could look up here. Obviously that means we need channel credentials. So future stuff.

Reply:
@voidzcy for the short term we don't need enum, it's only one keyword. For the long term we will see after channel credentials.

Reply:
Sure, then just keep using "google_default" (fyi, it is "google_default" instead of "compute_engine_creds", this is what being used in Go and C++) String. PTAL.